src/wallet/UserWalletTransactions.js

```diff
@@ -1,114 +1,169 @@
-    {transactions
-      .map((transaction, index) => {
-        const key = `${transaction.trx_id}${index}`;
-        const transactionType = transaction.op[0];
-        const transactionDetails = transaction.op[1];
+  static defaultProps = {
+    transactions: [],
+    currentUsername: '',
+  };
 
-        switch (transactionType) {
-          case 'transfer_to_vesting':
-            return (
-              <PowerUpTransaction
-                key={key}
-                amount={getFormattedTransactionAmount(transactionDetails.amount, 'SP')}
-                timestamp={transaction.timestamp}
-              />
-            );
-          case 'transfer':
-            if (transactionDetails.to === currentUsername) {
-              return (
-                <ReceiveTransaction
-                  key={key}
-                  from={transactionDetails.from}
-                  memo={transactionDetails.memo}
-                  amount={getFormattedTransactionAmount(transactionDetails.amount)}
-                  timestamp={transaction.timestamp}
-                />
-              );
-            }
-            return (
-              <TransferTransaction
-                key={key}
-                to={transactionDetails.to}
-                memo={transactionDetails.memo}
-                amount={getFormattedTransactionAmount(transactionDetails.amount)}
-                timestamp={transaction.timestamp}
-              />
-            );
-          case 'claim_reward_balance':
-            return (
-              <ClaimReward
-                key={key}
-                timestamp={transaction.timestamp}
-                rewardSteem={transactionDetails.reward_steem}
-                rewardSbd={transactionDetails.reward_sbd}
-                rewardVests={transactionDetails.reward_vests}
-                totalVestingShares={totalVestingShares}
-                totalVestingFundSteem={totalVestingFundSteem}
-              />
-            );
-          case 'transfer_to_savings':
-          case 'transfer_from_savings':
-          case 'cancel_transfer_from_savings':
-            return (
-              <SavingsTransaction
-                key={key}
-                transactionDetails={transactionDetails}
-                transactionType={transactionType}
-                amount={getFormattedTransactionAmount(transactionDetails.amount)}
-                timestamp={transaction.timestamp}
-              />
-            );
-          default:
-            return null;
+  constructor(props) {
+    super(props);
+    const firstTransaction = _.head(props.transactions);
+    this.state = {
+      userHasMoreTransactions: firstTransaction.actionCount > defaultAccountLimit,
+      lastActionCount: firstTransaction ? firstTransaction.actionCount : 0,
+      loadingMoreTransactions: false,
+    };
+  }
+
+  getFormattedTransactionAmount = (amount, currency) => {
+    const transaction = amount.split(' ');
+    const transactionAmount = parseFloat(transaction[0]).toFixed(3);
```

Comment:
Do we need `toFixed`? Wouldn't `FormattedNumber` do the same?

Reply:
yeah you're right, it's unnecessary here. good catch